### PR TITLE
ci: introduce a CI pipeline

### PR DESCRIPTION
**Pull request by wazelin**
# [TR-4672](https://oat-sa.atlassian.net/browse/TR-4672)

This PR aims to introduce a CI pipeline on TAO Community level so that _all_ extensions are checked on each release.

### CODE REVIEW

**Review by kilatib — APPROVED** [review]
LGTM

**Review by poyuki — APPROVED** [review]

**Comment by wazelin** [comment]
The fact that we have errors in the CI pipeline of the extensions in the release is a tad disturbing.

File: composer.json [hunk 1]
```diff
@@ -31,37 +31,43 @@
...
+    "oat-sa/extension-tao-clientdiag": "8.4.4",
+    "oat-sa/extension-tao-eventlog": "3.3.2",
+    "oat-sa/extension-tao-task-queue": "6.6.2",
+    "oat-sa/extension-tao-testqti-previewer": "3.7.2"
+  },
+  "require-dev": {
+    "mikey179/vfsstream": "~1",
+    "nimut/phpunit-merger": "~1",
+    "phpspec/prophecy": "~1",
+    "phpunit/phpunit": "^9.3"
```
Comment (gabrielfs7) [inline composer.json:70]: Will this be compatible with all extensions PHP unit?

Comment (wazelin) [reply composer.json:70]: They all already run under ~9

File: phpunit.xml.dist [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8"?>
+<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/9.3/phpunit.xsd"
+         bootstrap="./generis/test/bootstrap.php">
+    <testsuites>
+        <testsuite name="unit">
+            <directory>./*/test/unit</directory>
```
Comment (gabrielfs7) [inline phpunit.xml.dist:7]: Just a heads up that this is not the same path for all the repositories.

Comment (wazelin) [reply phpunit.xml.dist:7]: That's why we have a bit more elaborate logic within `./test.sh`

File: composer.json (hunk 1 above)
Comment (wazelin) [reply composer.json]: Darn, generis is still stuck on 8.5 indeed.
Extended the dependencies.

We'll have to update them sooner or later.

File: .github/workflows/continuous-integration.yaml [hunk 3]
```diff
@@ -0,0 +1,156 @@
+name: Continuous integration
+
+on:
+  push:
+    branches: [ master, release* ]
+  pull_request:
+    branches: [ master, release* ]
```
Comment (kilatib) [inline .github/workflows/continuous-integration.yaml:7]:
```suggestion
    branches: [ master, release*, feat* ]
```

Comment (wazelin) [reply .github/workflows/continuous-integration.yaml:7]: If there is a feature branch (for some reason) it must be merged via a PR.  No need to run _another_ job against the branch itself.

**Pull request merged**